ext/json/ext/simd/simd.h

```diff
@@ -18,6 +22,10 @@ typedef enum {
 
 static inline uint32_t trailing_zeros64(uint64_t input)
 {
+#ifdef JSON_DEBUG
+    assert(input > 0); // __builtin_ctz(0) is undefined behavior
```

Comment:
I'm doubting these `asserts` can actually fail. Not too sure what the compilation flags are on CI, but seems like they are optimized out even with  `JSON_DEBUG`. 

I'm unsure what's the best way to add assertions in ruby extensions. Perhaps I should just use `RUBY_ASSERT` and add a debug ruby in the CI matrix?